vscode/src/test/suite/ruby/chruby.test.ts

```diff
@@ -0,0 +1,160 @@
+
+const RUBY_VERSION = "3.3.0";
+
+suite("Chruby", () => {
+  if (os.platform() === "win32") {
+    // eslint-disable-next-line no-console
+    console.log("Skipping Chruby tests on Windows");
+    return;
+  }
+
+  let rootPath: string;
+  let workspacePath: string;
+  let workspaceFolder: vscode.WorkspaceFolder;
+  let outputChannel: WorkspaceChannel;
+
+  before(() => {
+    rootPath = fs.mkdtempSync(path.join(os.tmpdir(), "ruby-lsp-test-chruby-"));
+
+    fs.mkdirSync(path.join(rootPath, "opt", "rubies", RUBY_VERSION, "bin"), {
+      recursive: true,
+    });
+
+    // Create links to the real Ruby installations on CI and on our local machines
+    if (process.env.CI && os.platform() === "linux") {
+      fs.symlinkSync(
+        `/opt/hostedtoolcache/Ruby/${RUBY_VERSION}/x64/bin/ruby`,
+        path.join(rootPath, "opt", "rubies", RUBY_VERSION, "bin", "ruby"),
+      );
+    } else if (process.env.CI) {
+      fs.symlinkSync(
+        `/Users/runner/hostedtoolcache/Ruby/${RUBY_VERSION}/x64/bin/ruby`,
+        path.join(rootPath, "opt", "rubies", RUBY_VERSION, "bin", "ruby"),
+      );
+    } else {
+      fs.symlinkSync(
+        "/opt/rubies/3.3.0/bin/ruby",
+        path.join(rootPath, "opt", "rubies", RUBY_VERSION, "bin", "ruby"),
+      );
+    }
+
+    workspacePath = path.join(rootPath, "workspace");
+    fs.mkdirSync(workspacePath);
+
+    workspaceFolder = {
+      uri: vscode.Uri.from({ scheme: "file", path: workspacePath }),
+      name: path.basename(workspacePath),
+      index: 0,
+    };
+    outputChannel = new WorkspaceChannel("fake", LOG_CHANNEL);
+  });
+
+  after(() => {
+    fs.rmSync(rootPath, { recursive: true, force: true });
+  });
+
+  test("Finds Ruby when .ruby-version is inside workspace", async () => {
+    fs.writeFileSync(path.join(workspacePath, ".ruby-version"), RUBY_VERSION);
+
+    const chruby = new Chruby(workspaceFolder, outputChannel);
+    chruby.rubyInstallationUris = [
+      vscode.Uri.file(path.join(rootPath, "opt", "rubies")),
+    ];
+
+    const { env, version, yjit } = await chruby.activate();
+
+    assert.match(env.GEM_PATH!, /ruby\/3\.3\.0/);
+    assert.match(env.GEM_PATH!, /lib\/ruby\/gems\/3\.3\.0/);
+    assert.strictEqual(version, RUBY_VERSION);
+    assert.notStrictEqual(yjit, undefined);
+  });
+
+  test("Finds Ruby when .ruby-version is inside on parent directories", async () => {
+    fs.writeFileSync(path.join(rootPath, ".ruby-version"), RUBY_VERSION);
+
+    const chruby = new Chruby(workspaceFolder, outputChannel);
+    chruby.rubyInstallationUris = [
+      vscode.Uri.file(path.join(rootPath, "opt", "rubies")),
+    ];
+
+    const { env, version, yjit } = await chruby.activate();
+
+    assert.match(env.GEM_PATH!, /ruby\/3\.3\.0/);
+    assert.match(env.GEM_PATH!, /lib\/ruby\/gems\/3\.3\.0/);
+    assert.strictEqual(version, RUBY_VERSION);
+    assert.notStrictEqual(yjit, undefined);
+  });
+
+  test("Prefers standard rubies over custom built ones", async () => {
```

Comment:
Which part of the implementation handles this? (I couldn't see when searching for 'custom').

Reply:
It's actually just to prevent regressions. I had a previous implementation where the `engine-version` regex was incorrectly matching the custom part because it was something like `ruby-3.2.2-custom`.